### PR TITLE
Refactoring_mp_b2b

### PR DESCRIPTION
**Pull request by shubham-khandelwal2-metacube**
## PR Description

https://spryker.atlassian.net/browse/CC-16762

### CODE REVIEW

File: resources/environments/environments_api_mp_b2b.json [hunk 1]
```diff
@@ -273,22 +336,23 @@
...
-        "concrete_product_with_alternative_sku_name": "Verbatim USB stick Store n Go V3 49173 32GB USB3.0 gray",
-        "concrete_product_with_alternative_sku_name_in_lower_case": "verbatim usb stick store n go v3 49173 32gb usb3.0 gray",
-        "abstract_product_with_alternative_sku": "M21100",
-        "abstract_product_with_alternative_name": "Verbatim USB stick Store n Go V3 49173 32GB USB3.0 gray",
-        "alternative_abstract_product": "M21099",
-        "alternative_abstract_product_price_chf": "1130",
+        "concrete_product_with_alternative": {
+            "sku": "420566",
+            "sku_name": "Verbatim USB stick Store n Go V3 49173 32GB USB3.0 gray",
+            "sku_name_in_lower_case": "verbatim usb stick store n go v3 49173 32gb usb3.0 gray"
```
Comment (OKrivtsova) [inline resources/environments/environments_api_mp_b2b.json:342]: rename to name_lower_case

File: resources/environments/environments_api_mp_b2b.json [hunk 2]
```diff
@@ -273,22 +336,23 @@
...
+            "price_chf": "1130"
+        },
 
-        "abstract_available_product_with_stock": "M61125",
-        "abstract_available_product_with_stock_name": "Prosedia vertebral disc swivel chair with synchronous mechanics and contoured seat - backrest height 660 mm",
-        "abstract_available_product_with_stock_description": " Easy-to-care-for fabric cover. Substructure in graphite black RAL 9011.<br/>Standard rolling for soft floors.<br/>The backrest has height-adjustable lumbar support.  ",
-        "abstract_available_product_with_stock_superattribute": "bezugsfarbe",
-        "concrete_available_product_sku": "102712",
-        "concrete_available_product_meta_keywords": "Prosedia vertebral disc swivel chair with synchronous mechanics and contoured seat - backrest height 660 mm",
-        "concrete_available_product_meta_description": " Easy-to-care-for fabric cover. Substructure in graphite black RAL 9011.<br/>Standard rolling for soft floors.<br/>The backrest has height-adjustable lumbar support.  ",
```
Comment (OKrivtsova) [inline resources/environments/environments_api_mp_b2b.json:290]: why are these deleted and not replaced

File: resources/environments/environments_api_mp_b2b.json [hunk 3]
```diff
@@ -349,89 +413,115 @@
...
+            "volume_prices_name": "Post-it stick note Super Sticky Meeting Notes 6445-4SS 4 pieces/pack",
+            "original_prices": "M74972",
+            "label": "M21691",
+            "label_name": "Clairefontaine Collegeblock 8272C DIN A5, 90 sheets",
+            "concrete_offers": "",
+            "concrete_offers_name": "",
+            "reviews": "M719",
+            "reviews_name": "Rectangular table, rectangular tubing - 1600 x 800 mm",
+            "reviews_qty": "4",
+            "options": "M29501",
```
Comment (OKrivtsova) [inline resources/environments/environments_api_mp_b2b.json:434]: group to 
with_options:
sku, name

File: resources/environments/environments_api_mp_b2b.json [hunk 4]
```diff
@@ -349,89 +413,115 @@
...
-        "abstract_available_with_stock_and_never_out_of_stock": "M87",
         "concrete_available_with_stock_and_never_out_of_stock": "657716",
         "concrete_available_with_stock_and_never_out_of_stock_name": "FRIWA stackable chair - with closed back",
-        "abstract_available_product_with_no_stock_and_never_out_of_stock": "M90851",
-        "abstract_unavailable_product": "M1014948",
-        "concrete_unavailable_product_sku": "212447",
 
-        "abstract_available_product_with_3_concretes": "M29456",
-        "abstract_available_product_with_3_concretes_name": "Gmöhling aluminum data removal container - Exterior - LxWxH 435 x 385 x 650 mm",
-        "abstract_available_product_with_3_concretes_superattribute": "material",
```
Comment (OKrivtsova) [inline resources/environments/environments_api_mp_b2b.json:361]: group to 
abstract_product_with_variants:
"3_concretes": "M29456",
"3_concretes_name": "Gmöhling aluminum data removal container - Exterior - LxWxH 435 x 385 x 650 mm",
"3_concretes_superattribute": "material"

File: resources/environments/environments_api_mp_b2b.json [hunk 5]
```diff
@@ -7,200 +7,263 @@
...
+"shopping-list": {
         "shopping_list_name": "My Shopping List",
-        "yves_user_1st_shopping_list_id": "cf032865-d1ad-5e27-803a-423bd15ced66",
-        "yves_user_1st_shopping_list_name": "Newcomers",
+        "yves_user_1st_shopping_list": {
+            "id": "cf032865-d1ad-5e27-803a-423bd15ced66",
+            "name": "Newcomers"
+        },
         "yves_user_2nd_shopping_list_name": "Weekly office",
         "yves_user_3rd_shopping_list_name": "Monthly office"
```
[the remark this answers is not in the document]
Comment (shubham-khandelwal2-metacube) [reply resources/environments/environments_api_mp_b2b.json:157]: FIXED

File: resources/environments/environments_api_mp_b2b.json [hunk 6]
```diff
@@ -7,200 +7,263 @@
...
+        
     },
 
-    "search":{
+"search":{
         "total_number_of_products_in_search": "414",
         "default_ipp": "12",
         "ipp_middle": "24",
         "ipp_biggest": "36",
```
[the remark this answers is not in the document]
Comment (shubham-khandelwal2-metacube) [reply resources/environments/environments_api_mp_b2b.json:165]: FIXED

File: resources/environments/environments_api_mp_b2b.json [hunk 7]
```diff
@@ -7,200 +7,263 @@
...
+            "max": "5"
+        },
+        "label": {
+            "new": "New",
+            "sale": "SALE %",
+            "manual": "Top",
+            "discontinued": "Discontinued",
+            "alternatives": "Alternatives available"
+        },
+        "category_id": {
```
[the remark this answers is not in the document]
Comment (shubham-khandelwal2-metacube) [reply resources/environments/environments_api_mp_b2b.json:186]: FIXED

File: resources/environments/environments_api_mp_b2b.json [hunk 8]
```diff
@@ -7,200 +7,263 @@
...
+                "product_1_sku": "424453",
+                "product_1_name": "Schneider ballpoint pen HAPTIFY 135323 M 0.5 mm, dark blue/light blue",
+                "product_2_sku": "1001454",
+                "product_2_name": "Schomburg standing desk - HxWxD 1094 x 600 x 500 mm, beech",
+                "product_3_sku": "104517",
+                "product_3_name": "Wood stackable chair - 4-foot substructure, upholstered seat - Cover anthracite, substructure coated"                        
+                },
+            "qty_of_product": {
+                "in_bundle": "3",
+                "inside_bundle": "1"
```
[the remark this answers is not in the document]
Comment (shubham-khandelwal2-metacube) [reply resources/environments/environments_api_mp_b2b.json:223]: FIXED

File: resources/environments/environments_api_mp_b2b.json [hunk 9]
```diff
@@ -7,200 +7,263 @@
...
+                }
+        },
+        "product_abstract_sku": {
+            "multi_variant": "M58035",
+            "measurement_unit": "M23723",
+            "packaging_unit": "M21766"
```
[the remark this answers is not in the document]
Comment (shubham-khandelwal2-metacube) [reply resources/environments/environments_api_mp_b2b.json:229]: FIXED

File: resources/environments/environments_api_mp_b2b.json [hunk 10]
```diff
@@ -273,22 +336,23 @@
...
-        "concrete_product_with_alternative_sku": "420566",
-        "concrete_product_with_alternative_sku_name": "Verbatim USB stick Store n Go V3 49173 32GB USB3.0 gray",
-        "concrete_product_with_alternative_sku_name_in_lower_case": "verbatim usb stick store n go v3 49173 32gb usb3.0 gray",
-        "abstract_product_with_alternative_sku": "M21100",
-        "abstract_product_with_alternative_name": "Verbatim USB stick Store n Go V3 49173 32GB USB3.0 gray",
-        "alternative_abstract_product": "M21099",
-        "alternative_abstract_product_price_chf": "1130",
+        "concrete_product_with_alternative": {
+            "sku": "420566",
+            "sku_name": "Verbatim USB stick Store n Go V3 49173 32GB USB3.0 gray",
```
[the remark this answers is not in the document]
Comment (shubham-khandelwal2-metacube) [reply resources/environments/environments_api_mp_b2b.json:341]: FIXED

File: resources/environments/environments_api_mp_b2b.json (hunk 1 above)
Comment (shubham-khandelwal2-metacube) [reply resources/environments/environments_api_mp_b2b.json:342]: FIXED

File: resources/environments/environments_api_mp_b2b.json (hunk 3 above)
Comment (shubham-khandelwal2-metacube) [reply resources/environments/environments_api_mp_b2b.json:434]: FIXED

File: resources/environments/environments_api_mp_b2b.json [hunk 11]
```diff
@@ -349,89 +413,115 @@
...
-        "abstract_available_product_with_3_concretes": "M29456",
-        "abstract_available_product_with_3_concretes_name": "Gmöhling aluminum data removal container - Exterior - LxWxH 435 x 385 x 650 mm",
-        "abstract_available_product_with_3_concretes_superattribute": "material",
 
-        "abstract_product_with_volume_prices": "M21189",
-        "abstract_product_with_volume_prices_name": "Post-it stick note Super Sticky Meeting Notes 6445-4SS 4 pieces/pack",
-        "abstract_product_with_original_prices": "M74972",
-        "concrete_product_with_original_prices": "420687",
-        "concrete_product_with_original_prices_name": "Post-it Index Strong adhesive stripes, 686-F1EU 4 sheets sorted, 4 pieces/pack",
-        "concrete_product_with_volume_prices": "420685",
```
[the remark this answers is not in the document]
Comment (shubham-khandelwal2-metacube) [reply resources/environments/environments_api_mp_b2b.json:368]: FIXED

File: resources/environments/environments_api_mp_b2b.json [hunk 12]
```diff
@@ -349,89 +413,115 @@
             }
         },
 
-        "abstract_available_with_stock_and_never_out_of_stock": "M87",
         "concrete_available_with_stock_and_never_out_of_stock": "657716",
         "concrete_available_with_stock_and_never_out_of_stock_name": "FRIWA stackable chair - with closed back",
-        "abstract_available_product_with_no_stock_and_never_out_of_stock": "M90851",
-        "abstract_unavailable_product": "M1014948",
-        "concrete_unavailable_product_sku": "212447",
```
[the remark this answers is not in the document]
Comment (shubham-khandelwal2-metacube) [reply resources/environments/environments_api_mp_b2b.json:357]: FIXED

File: resources/environments/environments_api_mp_b2b.json (hunk 4 above)
Comment (shubham-khandelwal2-metacube) [reply resources/environments/environments_api_mp_b2b.json:361]: FIXED

File: resources/environments/environments_api_mp_b2b.json [hunk 13]
```diff
@@ -521,71 +625,98 @@
...
-        "payment_selection_name": "dummyMarketplacePaymentInvoice",
+        "payment": {
+            "provider_name": "dummyMarketplacePayment",
+            "method_name": "invoice",
+            "selection_name": "dummyMarketplacePaymentInvoice"
+        },
         "carrier_name": "DHL",
         "shipment_method_name": "Standard",
         "shipment_method_name_2": "Express",
         "delivery_date": "2030-12-31",
```
[the remark this answers is not in the document]
Comment (shubham-khandelwal2-metacube) [reply resources/environments/environments_api_mp_b2b.json:636]: FIXED

File: resources/environments/environments_api_mp_b2b.json (hunk 2 above)
Comment (shubham-khandelwal2-metacube) [reply resources/environments/environments_api_mp_b2b.json:290]: these already replaced by "abstract_available_product_with_stock" group.

File: resources/environments/environments_api_mp_b2b.json [hunk 14]
```diff
@@ -349,89 +413,115 @@
...
+"abstract_product": {
+        "abstract_unavailable_product": "M1014948",
+        "abstract_product_with": {
+            "volume_prices": "M21189",
+            "volume_prices_name": "Post-it stick note Super Sticky Meeting Notes 6445-4SS 4 pieces/pack",
+            "original_prices": "M74972",
+            "label": "M21691",
+            "label_name": "Clairefontaine Collegeblock 8272C DIN A5, 90 sheets",
+            "concrete_offers": "",
+            "concrete_offers_name": "",
```
[the remark this answers is not in the document]
Comment (shubham-khandelwal2-metacube) [reply resources/environments/environments_api_mp_b2b.json:430]: DONE

File: resources/environments/environments_api_mp_b2b.json [hunk 15]
```diff
@@ -7,200 +7,263 @@
...
+                "in_bundle": "3",
+                "inside_bundle": "1"
+                }
+        },
+        "product_abstract_sku": {
+            "multi_variant": "M58035",
+            "measurement_unit": "M23723",
+            "packaging_unit": "M21766"
+        },
+        "product_with_relations": {
```
[the remark this answers is not in the document]
Comment (shubham-khandelwal2-metacube) [reply resources/environments/environments_api_mp_b2b.json:231]: FIXED

File: resources/environments/environments_api_mp_b2b.json [hunk 16]
```diff
@@ -349,89 +413,115 @@
...
-        "concrete_product_random_weight_amount": "5.0000000000",
-        "concrete_product_random_weight_conversion": "1",
-        "concrete_product_random_weight_precision": "100",
-
-        "concrete_product_random_weight_packaging_unit": "19",
         "measurement_unit_m": "METR",
 
         "product_option_1": "OP_2_year_waranty",
         "product_option_1_name": "Two (2) year limited warranty",
         "product_option_2": "OP_insurance",
```
[the remark this answers is not in the document]
Comment (shubham-khandelwal2-metacube) [reply resources/environments/environments_api_mp_b2b.json:491]: FIXED

File: resources/environments/environments_api_mp_b2b.json (hunk 4 above)
Comment (shubham-khandelwal2-metacube) [reply resources/environments/environments_api_mp_b2b.json:361]: I was getting error "failed: SyntaxError: invalid decimal literal (<string>, line 1)" if any variables starts with numbers so I changed these variables names to :
concretes_3": 
concretes_name_3": 
concretes_superattribute_3":

File: resources/environments/environments_api_mp_b2b.json [hunk 17]
```diff
@@ -6,259 +6,365 @@
...
+                "abstract_sku": "M21100"
+                },
+            "has_upselling_products": {
+                "abstract_sku": "M29502",
+                "concrete_sku": "419880"
+                },
+            "product_related_product_with_upselling_relation_sku": "M29499",
+            "product_related_product_with_upselling_relation_name": "Safescan counterfeit test device - SAFESCAN 45 UV",
+            "product_related_product_with_related_relation_sku": "M29501",
+            "product_related_product_with_related_relation_name": "Safescan automatic banknote counter - UV counterfeit recognition"
```
Comment (OKrivtsova) [inline resources/environments/environments_api_mp_b2b.json:287]: these 4 should also be grouped  - maybe it makes sense to add them to has_upselling_products and has_related_products  but so that we know which is which - or group separately

**Review by OKrivtsova — COMMENTED** [review]
please for the next review attach the result of running all the tests (make sure that all fails have bugs assigned)

**Comment by shubham-khandelwal2-metacube** [comment]
[mp_b2b_results.zip](https://github.com/spryker/robotframework-suite-tests/files/8918572/mp_b2b_results.zip)

**Review by OKrivtsova — COMMENTED** [review]
I do not see open comments answered
Also from the linked report - there are fails that do not have bugs associated with them that seem to be reelated to the variables. e.g.
Get_concrete_bundled_products_inside_concrete_bundle
Get_concrete_bundled_products_inside_concrete_bundle_with_included_concretes
Get_concrete_bundle_product_with_bundled_products_include
Get_abstract_bundle_product_with_bundled_products_include
Get_abstract_product_with_product_labels_included

File: resources/environments/environments_api_mp_b2b.json (hunk 17 above)
Comment (shubham-khandelwal2-metacube) [reply resources/environments/environments_api_mp_b2b.json]: DONE

File: tests/api/mp_b2b/glue/cart_endpoints/shared-carts/positive.robot [hunk 18]
```diff
@@ -119,7 +119,7 @@ Add_an_item_to_the_shared_shopping_cart_by_user_with_access
     ...    AND    Save value to a variable:    [data][0][id]    companyUserId
     ...    AND    I send a POST request:    /carts/${cartId}/shared-carts    {"data":{"type":"shared-carts","attributes":{"idCompanyUser":"${companyUserId}","idCartPermissionGroup":2}}}
     ...    AND    Response status code should be:    201
-    ...    AND    Then I get access token for the customer:    ${yves_shared_shopping_cart_user_email}
+    ...    AND    I get access token for the customer:    ${yves_shared_shopping_cart_user_email}
```
[the remark this answers is not in the document]
Comment (shubham-khandelwal2-metacube) [reply tests/api/mp_b2b/glue/cart_endpoints/shared-carts/positive.robot]: DONE

File: tests/api/mp_b2b/glue/cart_endpoints/shared-carts/positive.robot [hunk 19]
```diff
@@ -119,7 +119,7 @@ Add_an_item_to_the_shared_shopping_cart_by_user_with_access
...
     ...    AND    I set Headers:    Authorization=${token}
     When I send a POST request:    /carts/${cartId}/items?include=items    {"data":{"type":"items","attributes":{"sku":"${concrete_available_product_sku}","quantity":1}}}
```
[the remark this answers is not in the document]
Comment (shubham-khandelwal2-metacube) [reply tests/api/mp_b2b/glue/cart_endpoints/shared-carts/positive.robot]: FIXED

File: tests/api/mp_b2b/glue/cart_endpoints/shared-carts/positive.robot [hunk 20]
```diff
@@ -144,7 +144,7 @@ Update_an_item_quantity_at_the_shared_shopping_cart_with_full_access_permissions
     ...    AND    Save value to a variable:    [data][0][id]    companyUserId
```
[the remark this answers is not in the document]
Comment (shubham-khandelwal2-metacube) [reply tests/api/mp_b2b/glue/cart_endpoints/shared-carts/positive.robot:144]: DONE

File: tests/api/mp_b2b/glue/cart_endpoints/shared-carts/positive.robot [hunk 21]
```diff
@@ -170,7 +170,7 @@ Delete_an_item_from_the_shared_shopping_cart_with_full_access_permissions_by_use
...
     ...    AND    I send a POST request:    /carts/${cartId}/shared-carts    {"data":{"type":"shared-carts","attributes":{"idCompanyUser":"${companyUserId}","idCartPermissionGroup":2}}}
     ...    AND    Response status code should be:    201
-    ...    AND    Then I get access token for the customer:    ${yves_shared_shopping_cart_user_email}
+    ...    AND    I get access token for the customer:    ${yves_shared_shopping_cart_user_email}
```
[the remark this answers is not in the document]
Comment (shubham-khandelwal2-metacube) [reply tests/api/mp_b2b/glue/cart_endpoints/shared-carts/positive.robot]: DONE

**Review by OKrivtsova — APPROVED** [review]
